File: platforms/android/stato/src/main/java/com/facebook/stato/nativeplugins/components/Sidebar.java
package com.facebook.stato.nativeplugins.components;

import com.facebook.stato.core.StatoArray;

public class Sidebar {

  private final StatoArray.Builder sections = new StatoArray.Builder();

  public Sidebar addSection(SidebarSection section) {
    sections.put(section.serialize());
    return this;
  }

  public StatoArray serialize() {
    return sections.build();
  }
}

File: platforms/android/android-stato/src/main/kotlin/org/stato/nativeplugins/components/Sidebar.kt
package org.stato.nativeplugins.components

import org.stato.core.StatoArray

class Sidebar {

  private val sections = StatoArray.Builder()

  fun addSection(section: SidebarSection): Sidebar {
    sections.put(section.serialize())
    return this
  }

  fun serialize(): StatoArray {
    return sections.build()
  }
}
